### PR TITLE
add service target, field's required and selector properties to HassService model

### PR DESCRIPTION
**Pull request by makp0**

### CODE REVIEW

File: src/HassClient/Helpers/Json/JsonExtensions.cs [hunk 1]
```diff
@@ -132,27 +143,60 @@ HassService getServiceFields(string service, JsonElement element)
...
+            object? getSelector(string selectorName, JsonElement element)
+            {
+                var selectorType = executingAssemblyTypes.FirstOrDefault(x => string.Equals($"{selectorName}Selector", x.Name, StringComparison.OrdinalIgnoreCase));
+
+                if (selectorType is null)
+                {
+                    return null;
+                }
+
+                if (element.ValueKind == JsonValueKind.Object && !element.EnumerateObject().Any() ||
```
Comment (FrankBakkerNl) [inline src/HassClient/Helpers/Json/JsonExtensions.cs:179]: So If I understand correct
selector=  { Text = null}
Will result in The Text property != null

File: src/HassClient/Helpers/Json/JsonExtensions.cs [hunk 2]
```diff
@@ -124,6 +133,8 @@ HassService getServiceFields(string service, JsonElement element)
                 var serviceFields = new List<HassServiceField>();
```
Comment (FrankBakkerNl) [inline src/HassClient/Helpers/Json/JsonExtensions.cs:133]: Ah, so the current model was already deserialized manually for the most part

File: src/HassClient/Model/Selectors.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        [Required]
+        public double Max { get; init; }
+
+        public float? Step { get; init; }
+
+        public string? UnitOfMeasurement { get; init; }
+
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public NumberSelectorMode? Mode { get; init; }
```
Comment (FrankBakkerNl) [inline src/HassClient/Model/Selectors.cs:60]: Could this break if HA adds a new mode later on, or will it have null?

**Review by FrankBakkerNl — APPROVED** [review]

File: src/HassClient/Helpers/Json/JsonExtensions.cs (hunk 1 above)
Comment (makp0) [reply src/HassClient/Helpers/Json/JsonExtensions.cs:179]: yes

File: src/HassClient/Helpers/Json/JsonExtensions.cs (hunk 2 above)
Comment (makp0) [reply src/HassClient/Helpers/Json/JsonExtensions.cs:133]: yes

**Comment by makp0** [comment]
closed the pull request so I can reopen the one with fixed warnings and without required field. making it more atomic and clean

File: src/HassClient/Model/Selectors.cs (hunk 3 above)
Comment (makp0) [reply src/HassClient/Model/Selectors.cs:60]: it will break, unfortunately. 
looking for a fix

**Pull request closed without merging**